From: Ben Hutchings <ben@decadent.org.uk>
Date: Mon, 23 May 2016 01:25:47 +0100
Subject: net/sched: Fix ABI change in 4.5.5
Forwarded: not-needed

Restore the function qdisc_tree_decrease_qlen(), removed in 4.5.5.
It can now be a trivial wrapper for its replacement,
qdisc_tree_reduce_backlog().

---
--- a/include/net/sch_generic.h
+++ b/include/net/sch_generic.h
@@ -398,6 +398,7 @@ void qdisc_reset(struct Qdisc *qdisc);
 void qdisc_destroy(struct Qdisc *qdisc);
 void qdisc_tree_reduce_backlog(struct Qdisc *qdisc, unsigned int n,
 			       unsigned int len);
+void qdisc_tree_decrease_qlen(struct Qdisc *qdisc, unsigned int n);
 struct Qdisc *qdisc_alloc(struct netdev_queue *dev_queue,
 			  const struct Qdisc_ops *ops);
 struct Qdisc *qdisc_create_dflt(struct netdev_queue *dev_queue,
--- a/net/sched/sch_api.c
+++ b/net/sched/sch_api.c
@@ -782,6 +782,12 @@ void qdisc_tree_reduce_backlog(struct Qd
 }
 EXPORT_SYMBOL(qdisc_tree_reduce_backlog);
 
+void qdisc_tree_decrease_qlen(struct Qdisc *sch, unsigned int n)
+{
+	qdisc_tree_reduce_backlog(sch, n, 0);
+}
+EXPORT_SYMBOL(qdisc_tree_decrease_qlen);
+
 static void notify_and_destroy(struct net *net, struct sk_buff *skb,
 			       struct nlmsghdr *n, u32 clid,
 			       struct Qdisc *old, struct Qdisc *new)
